src/core/packet.lua

```diff
@@ -45,18 +47,25 @@ end
 
 -- Freelist containing empty packets ready for use.
 
-ffi.cdef[[
+local max_packets = 1e6
+
+ffi.cdef([[
 struct freelist {
+    int32_t lock[1];
     uint64_t nfree;
     uint64_t max;
-    struct packet *list[?];
+    struct packet *list[]]..max_packets..[[];
```

Comment:
`[?]` doesn’t work because this branch maps it as a SHM object and `core.shm` would then fail to determine the `sizeof` this ctype. So I guess the remaing purpose of `max_packets` is to give this value a name in the source.

Reply:
What I mean to say is that you could have:

```c
enum { MAX_PACKETS=1000000 };
struct freelist {
  int32_t lock[1];
  uint32_t nfree;
  struct packet *list[MAX_PACKETS};
};
```

That way you could still access max_packets by ffi.C.MAX_PACKETS, but since we are really *requiring* that freelists have a fixed size, we do away with the notion that the freelist size is configurable.  Also that uses uint32_t for the occupancy for better struct packing.  WDYT?

Or, if preserving variable-sized freelists is important (probably it is), then we need to use some other variable-size-friendly core.shm mapper.